libraries/chain/account_evaluator.cpp

```diff
@@ -435,6 +435,7 @@ void_result account_upgrade_evaluator::do_apply(const account_upgrade_evaluator:
          // Upgrade to lifetime member. I don't care what the account was before.
          a.statistics(d).process_fees(a, d);
          a.membership_expiration_date = time_point_sec::maximum();
+         a.original_registrar = a.registrar;
```

Comment:
probably best to do the exact same thing with `referrer`.

Reply:
Okay will add that too.